x-pack/test/functional/apps/rollup_job/hybrid_index_pattern.js

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+import datemath from '@elastic/datemath';
+import expect from '@kbn/expect';
+import mockRolledUpData, { mockIndices } from './hybrid_index_helper';
+
+export default function ({ getService, getPageObjects }) {
+
+  const es = getService('es');
+  const esArchiver = getService('esArchiver');
+  const retry = getService('retry');
+  const PageObjects = getPageObjects(['security', 'rollup', 'common', 'indexManagement', 'settings', 'discover']);
+
+  describe('hybrid index pattern', function () {
+    const rollupJobName = 'rollup-to-be-' + Math.floor(Math.random() * 10000);
+    const indexName = 'rollup-to-be';
+    const now = new Date();
+    const pastDates = [
+      datemath.parse('now-1d', { forceNow: now }),
+      datemath.parse('now-2d', { forceNow: now }),
+      datemath.parse('now-3d', { forceNow: now }),
+    ];
+
+    it('create hybrid index pattern', async () => {
+      //Create data for rollup job to recognize.
+      //Index past data to be used in the test.
+      await pastDates.map(async (day) => {
+        await es.index(mockIndices(day));
+      });
+
+      await retry.waitForWithTimeout('waiting for 3 records to be loaded into elasticsearch.', 10000, async () => {
+        const response = await es.indices.get({
+          index: 'to-be*',
+          allow_no_indices: false
+        });
+        return Object.keys(response).length === 3;
+      });
+
+      await retry.try(async () => {
```

Comment:
Just to make sure I'm following, it looks like this code:

1. Creates `to-be-rolled-up` indices on line 33.
2. Creates a `rollup-to-be-{random number}` rollup job on line 46 which captures the indices matching `to-be*` and rolls them up into a `rollup-to-be` index.
3. Populates the `rollup-to-be` index with fake rollup data on line 68.
4. Creates `live-data` indices on line 81.
5. Asserts that the rollup index pattern option is available and can be used to create an index pattern out of the `live-data` indices and the `rollup-to-be` index on line 85.
6. Asserts that Discover can retrieve and render results from this index pattern on line 92.

I have a couple suggestions here. To make the code easier to follow I'd suggest using variables and descriptive names to clarify all of the indices and the rollup job in play. For example:

```js
const rollupSourceIndexPrefix = `rollupSourceData`; // Replaces `to-be-rolled-up`. Pass this into mockIndices(day, rollupSourceIndexPrefix)
const rollupJobName = `hybridIndexPatternTestRollupJob`;
const rollupTargetIndexName = `rollupTargetData`; // Replaces `rollup-to-be`
const regularIndexPrefix = `regularIndex`; // Replaces `live-data`
```

I'd also suggest reconsidering the need for `live-data` and the assertion against Discover. This is really just testing that the search strategy for retrieving data from this type of index pattern works correctly, which is unrelated to the logic for creating a rollup index pattern. I think we should keep this test focused on just testing the ability to create a rollup index pattern, and investigate better ways of testing the logic in Discover (e.g. unit tests for the underlying search strategy).

Reply:
I understand the concern about asserting in discover against the hit count since that doesn't necessarily include our UI functionality. But I still need live data to create the hybrid indeed pattern and then we can validate it by checking the table for it.

Reply:
Can you create a hybrid index pattern using the source indices?

> we can validate it by checking the table for it

The table of index patterns under Management > Index Patterns, right? 👍

Reply:
Yes.

Reply:
The source Indices go into the rollup. The hybrid is a mixture of rolled up data and live data so that's why I add the additional live data.

Reply:
I understand. My point is that the hybrid index pattern doesn't care about how the indices it captures are being used, as long as some of them are standard indices and some of them are rollup indices -- right? In that case, why can't we reuse the source indices? If we can reuse the source indices then we don't need to bother creating the `live-data` indices.

Reply:
Fixed by checking if the indexPattern table isn't empty per Kibana QA.